lib/improver/weighted_blend.py

```diff
@@ -33,13 +33,217 @@
+        """Represent the configured plugin instance as a string."""
+        result = ('<BlendingUtilities>')
+        return result
+
+    @staticmethod
+    def basic_weighted_average(cube, weights, coord, coord_dim):
+        """Calculate weighted mean across the chosen coord
+
+        Args:
+            cube : iris.cube.Cube
+                   Cube to blend across the coord.
+            weights: Optional list or np.array of weights
+                     or None (equivalent to equal weights)
+            coord : string
+                     The name of a coordinate dimension in the cube.
+            coord_dim : tuple
+                     The index of the coordinate dimension in the cube.
+
+        Returns:
+            result : iris.cube.Cube
+                     containing the weighted mean across the chosen coord
+        """
+        # Supply weights as an array of weights whose shape matches the cube.
+        weights_array = None
+        if weights is not None:
+            weights_array = iris.util.broadcast_to_shape(np.array(weights),
+                                                         cube.shape,
+                                                         coord_dim)
+        # Calculate the weighted average.
+        result = cube.collapsed(coord,
+                                iris.analysis.MEAN, weights=weights_array)
+        return result
+
+    @staticmethod
+    def blend_percentile_cube(cube, weights, coord, coord_dim, perc_coord):
+        """ Blend together percentile cube
+
+        Args:
+            cube : iris.cube.Cube
+                   Cube to blend across the coord.
+            weights: Optional list or np.array of weights
+                     or None (equivalent to equal weights)
+            coord : string
+                     The name of a coordinate dimension in the cube.
+            coord_dim : tuple
+                     The index of the coordinate dimension in the cube.
+            perc_coord : iris.cube.DimCoord
+                     The perecentile coordinate
+
+        Returns:
+            result : iris.cube.Cube
+                     containing the weighted percentile blend
+                     across the chosen coord
+        """
+
+        percentiles = np.array(perc_coord.points, dtype=float)
+        num = cube.data.shape[coord_dim[0]]
+        if weights is None:
+            weights = np.ones(num)/float(num)
+        PERCENTILE_BLEND = (Aggregator('percentile_blend',
+                            BlendingUtilities.blend_percentile_aggregate))
+        perc_dim, = cube.coord_dims(perc_coord.name())
+
+        result = cube.collapsed(coord,
+                                PERCENTILE_BLEND,
+                                arr_percent=percentiles,
+                                arr_weights=weights,
+                                perc_dim=perc_dim)
+
+        # Add meta data
+        new_name = 'over {}'.format(coord)
+        result.attributes.update({'Blended': new_name})
```

Comment:
For now I have taken this meta data out as it wasn't obvious from the CF document. There is still a new cell method percentile_blend which iris adds but which isn't part of the CF standard. I'm going to leave it in for now and have updated our internal documents to show that it has been added.